test/set.js

```diff
@@ -22,6 +22,22 @@ describe('set', function() {
     assert.deepEqual(b, {w: 1, x: {a: 5, b: 2, c: 3}, z: 4});
   });
 
+  it('handles empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({}, a);
+    assert.deepEqual(b, a);
+  });
+
+  it('handles nested empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({w: 0, x: {}, z: 3}, a);
+    assert.deepEqual(b, {w: 0, x: {a: 1, b: 2, c: 3}, z: 3});
+  });
+
```

Comment:
This feels like a naming issue. What I think of as `set` is called `assoc` and `assocPath`, and what I think of as `merge` is now being called `set`? What was wrong with the name `merge`? I would definitely expect an operation called `set` to result in `{w: 0, x: {}, z: 3}`. And `merge` resulting in `{w: 0, x: {a: 1, b: 2, c: 3}, z: 3}` seems reasonable, especially with docs/examples/tests so everyone is aware of that behaviour.